src/FolderConfig.cpp

```diff
@@ -0,0 +1,54 @@
+#include "FolderConfig.h"
+#include <QFile>
+#include <QDebug>
+#include<QDir>
+
+namespace QtDBMigrationNS {
+FolderConfig::FolderConfig(const QString &path): AbstractConfig(path)
+{
+}
+
+bool FolderConfig::load()
+{
+    QDir dir(m_path);
+    QStringList nameFilter("*.sql");
+    QStringList sqlFiles = dir.entryList(nameFilter);
+    int totalFiles = sqlFiles.count();
+
+    if (!dir.exists() || totalFiles < 1) {
+        return false;
+    }
+
+    foreach(QString filename, sqlFiles) {
+        QString filePath = dir.filePath(filename);
+        QFile file(filePath);
+        if (!file.exists()) {
+            return false;
+        }
+
+        QFileInfo fileInfo(file.fileName());
+        QString fileName(fileInfo.baseName());
+        QRegExp rx("[0-9][0-9][0-9][0-9]");
+        if (!rx.exactMatch(fileName)) {
+            continue;
+        }
+
+        if(!file.open(QFile::ReadOnly | QIODevice::Text)) {
+            return false;
+        }
+
+
+        QString fileContents = file.readAll();
+        QList<QString> fileSplit = fileContents.split("\n", Qt::SplitBehavior(Qt::SplitBehaviorFlags::SkipEmptyParts));
+        QString applySql = fileSplit[0];
+        QString revertSql = fileSplit[1];
+
+
+        m_schemaVersions << new SchemaVersion(
+                            file.fileName().toInt(), applySql, revertSql);
```

Comment:
Filename should have an extension.

Also, you can use `QRegExp` to extract number from whole filename.

Reply:
Changed to using regex.